Bio/motifs/xms.py

```diff
@@ -0,0 +1,95 @@
+# choice of the "Biopython License Agreement" or the "BSD 3-Clause License".
+# Please see the LICENSE file that should have been included as part of this
+# package.
+
+"""Parse XMS motif files."""
+
+from Bio import motifs
+from Bio.Alphabet import IUPAC
+
+
+from xml.dom import minidom, Node
+import re
+
+
+class XMSScanner:
+    def __init__(self, doc):
+        """Generate motif Record from xms document, an XML-like motif pfm file."""
+        self.record = Record()
+        for child in doc.getElementsByTagName('motif'):
+            if child.nodeType == Node.ELEMENT_NODE:
+                self.handle_motif(child)
+
+    def handle_motif(self, node):
+        motif_name = self.get_text(node.getElementsByTagName("name"))
+        nucleotide_counts = {'A': [], 'C': [], 'G': [], 'T': []}
+
+        for column in node.getElementsByTagName("column"):
+            [nucleotide_counts[nucleotide].append(float(nucleotide_count))
+             for nucleotide, nucleotide_count in zip(['A', 'C', 'G', 'T'], self.get_acgt(column))]
+
+        motif = motifs.Motif(alphabet=IUPAC.unambiguous_dna, counts=nucleotide_counts)
+        motif.name = motif_name
+
+        self.record.append(motif)
+
+    def get_property_value(self, node, key_name):
+        for cur_property in node.getElementsByTagName('prop'):
+            right_property = False
+            cur_value = None
+            for child in cur_property.childNodes:
+                if child.nodeType != Node.ELEMENT_NODE:
+                    continue
+                if child.tagName == 'key' and self.get_text([child]) == key_name:
+                    right_property = True
+                if child.tagName == 'value':
+                    cur_value = self.get_text([child])
+            if right_property:
+                return cur_value
+        return None
+
+    def get_acgt(self, node):
+        a, c, g, t = 0.0, 0.0, 0.0, 0.0
+        for weight in node.getElementsByTagName('weight'):
+            if weight.getAttribute("symbol") == "adenine":
+                a = float(self.get_text([weight]))
+            elif weight.getAttribute("symbol") == "cytosine":
+                c = float(self.get_text([weight]))
+            elif weight.getAttribute("symbol") == "guanine":
+                g = float(self.get_text([weight]))
+            elif weight.getAttribute("symbol") == "thymine":
+                t = float(self.get_text([weight]))
+        return a, c, g, t
+
+    def get_text(self, nodelist):
+        retlist = []
+        for node in nodelist:
+            if node.nodeType == Node.TEXT_NODE:
+                retlist.append(node.wholeText)
+            elif node.hasChildNodes:
+                retlist.append(self.get_text(node.childNodes))
+
+        return re.sub('\s+', ' ', ''.join(retlist))
+
+
+class Record(list):
+    """A Bio.motifs.xms.Record stores the information in a XMS matrix table.
+    The record inherits from a list containing the individual motifs.
```

Comment:
That needs a blank line to separate the one-line summary from the rest of the docstring, see https://github.com/biopython/biopython/issues/1170

(We now enforce this via TravisCI/Tox)

Reply:
Fixed. I don't know if I'm going to have time to add a section to the docs about this. A line or two saying to look at the API docs for `pfm.py` might be a good idea, as it should provide at least basic functionality for most common formats.